samples/bluetooth/peripheral_uart/src/main.c

```diff
@@ -377,6 +424,44 @@ void error(void)
 	}
 }
 
+static void num_comp_reply(struct bt_conn *conn, bool accept)
+{
+	if (accept) {
+		bt_conn_auth_passkey_confirm(conn);
+		printk("Numeric Match, conn %p\n", conn);
+	} else {
+		bt_conn_auth_cancel(conn);
+		printk("Numeric Reject, conn %p\n", conn);
+	}
+}
+
+void button_changed(u32_t button_state, u32_t has_changed)
+{
+	u32_t buttons = button_state & has_changed;
+
+	if (auth_conn) {
+		if (buttons & KEY_PASSKEY_ACCEPT) {
+			num_comp_reply(auth_conn, true);
+		}
+
+		if (buttons & KEY_PASSKEY_REJECT) {
+			num_comp_reply(auth_conn, false);
+		}
+
+		bt_conn_unref(auth_conn);
```

Comment:
To support more buttons you should move this into num_comp_reply

Reply:
Ok. Moved